keyboards/hackpad/keymaps/default/keymap.c

```diff
@@ -0,0 +1,39 @@
+// Copyright 2023 Nico Stuhlmueller (@ThePurox)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+#include "keymap_german.h"
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /*             ┌───┐
+     *             │MUT│
+     * ┌───┬───┬───┼───┤
+     * │ 7 │ 8 │ 9 │ + │
+     * ├───┼───┼───┼───┤
+     * │ 4 │ 5 │ 6 │ - │
+     * ├───┼───┼───┼───┤
+     * │ 1 │ 2 │ 3 │ * │
+     * ├───┼───┼───┼───┤
+     * │ 0 │ . │Ent│ / │
+     * └───┴───┴───┴───┘
+     */
+    [0] = LAYOUT_numpad_4x4(                      LT(1, KC_MUTE),
+                            KC_7, KC_8,   KC_9,   DE_PLUS,
+                            KC_4, KC_5,   KC_6,   DE_MINS,
+                            KC_1, KC_2,   KC_3,   DE_ASTR,
+                            KC_0, KC_DOT, KC_ENT, DE_SLSH),
+
+    [1] = LAYOUT_numpad_4x4(                              XXXXXXX,
+                            RGB_M_P,  RGB_M_B,  RGB_M_R,  RGB_HUI,
+                            RGB_M_SW, RGB_M_SN, RGB_M_K,  RGB_HUD,
+                            RGB_M_X,  RGB_M_G,  RGB_M_TW, RGB_VAI,
+                            RGB_TOG,  RGB_SAI,  RGB_SAD,  RGB_VAD)};
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (clockwise) {
+        tap_code_delay(KC_VOLU, 10);
+    } else {
+        tap_code_delay(KC_VOLD, 10);
+    }
+    return false;
+}
```

Comment:
The default encoder behavior needs to be implemented at the keyboard level (implement `encoder_update_kb()` in `keyboards/hackpad/hackpad.c`); this is required to make encoders do something in configurator builds (the configurator does not have encoder support at the moment, so it relies on `encoder_update_kb()` being implemented at the keyboard level).

However, you may consider using `ENCODER_MAP_ENABLE = yes` in keymaps (but enabling that option at the keyboard level is not allowed).  This is also the only allowed way to support encoders in VIA, if you choose to add VIA support to your board.

Reply:
moved it